tests/functional/behave_features/common/utils/workflow_repo_manager.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+"""Utility class for managing local/remote git branches and worktrees"""
+
+import os
+import logging
+import git
+
+from tempfile import TemporaryDirectory
+
+import common.utils.github as github
+
+
+class WorkflowRepoManager:
+    # Keep a log of things created so we can clean them up.
+    __local_branches_created: [str] = []
+    __local_worktrees_created: [TemporaryDirectory] = []
+    # (remote, branch), e.g. ('openshift-helm-charts/charts, 'my-pr-branch')
+    __remote_branches_created: [(str, str)] = []
+
+    # The token to use for GitHub API operations.
+    __authtoken: str = ""
+
+    # The branch at repo initialization. On Cleanup, we return to this branch
+    # before we remove locally generated branches.
+    original_branch: str = None
+
+    # Working directory at instantiation.
+    old_cwd: str = os.getcwd()
+
+    # The repository at working directory.
+    repo: git.Repo = None
+
+    def __init__(self):
+        logging.debug(f"{self} --> __init__ called!")
+
+        try:
+            self.repo = git.Repo()
+        except Exception as e:
+            raise Exception(
+                f"Expected to be able to initialize a git repository at this path and failed: {e}"
+            )
```

Comment:
I'm not sure to which comment this refers. The GitHub UI shuffles things around, I think, in between your code review and what it presents to me.

Reply:
This should be addressed. Initialization can raise two exceptions. The first is accounted for, but the second (NoSuchPath) is not because we should never encounter this one at the time this code is reached. Unless I'm mistaken, we must pass a path other than the current working directory to hit NoSuchPath, and we don't do that as far as I can see.